src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,35 @@
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save movie to DB : " + movie.getTitle(), e
+            );
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Movie movie = session.get(Movie.class, id);
+            session.close();
+            return Optional.ofNullable(movie);
+        } catch (RuntimeException e) {
+            throw new DataProcessingException("Can't get a movie wih id : " + id);
```

Comment:
The `DataProcessingException` constructor call in the catch block of the `get` method is missing the cause exception `e`. Consider adding it to provide more context for the exception: `throw new DataProcessingException("Can't get a movie with id : " + id, e);`